src/core/qgsfield.cpp

```diff
@@ -589,12 +589,61 @@ bool QgsField::convertCompatible( QVariant &v, QString *errorMessage ) const
     return false;
   }
 
+  if ( d->typeName == "geometry" && v.type() == QVariant::String )
+  {
+    // TODO: expose this as a "canonical text input" method of
```

Comment:
Is this part could be useful in a dedicated method?

Reply:
Yes, I think "canonical text input and output" methods should be exposed for the ReferencedGeometry class